lib/libc/math/lib_log.c

```diff
@@ -51,17 +51,34 @@
 #include <math.h>
 #include <float.h>
 
+/************************************************************************
+ * Pre-processor Definitions
+ ************************************************************************/
+
+/* To avoid looping forever in particular corner cases, every LOGF_MAX_ITER
+ * the error criteria is relaxed by a factor LOGF_RELAX_MULTIPLIER.
+ * todo: might need to adjust the double floating point version too.
+ */
+
+#define LOGF_MAX_ITER         10
+#define LOGF_RELAX_MULTIPLIER 2
 /************************************************************************
  * Public Functions
  ************************************************************************/
 
+/****************************************************************************
+ * Name: log
+ ****************************************************************************/
+
 #ifdef CONFIG_HAVE_DOUBLE
 double log(double x)
 {
 	double y;
 	double y_old;
-	double ey;
+	double ney;
 	double epsilon;
+	int    relax_factor;
+	int    iter;
 
 	if (x < 0.0) {
```

Comment:
I think, x cannot be 0.0 also.

Reply:
As this is the original code, I will update it in next Pull Request.